test/distributed/test_nvshmem.py

```diff
@@ -65,6 +65,20 @@ def foo():
         out = symm_mem.empty(numel, dtype=dtype, device=self.device)
         symm_mem.rendezvous(out, group=group_name)
 
+    @skipIfRocm
+    def test_alloc_without_device_context(self) -> None:
+        # NOTE: required for nvshmem allocation
+        torch.empty(1, device=self.device)
```

Comment:
why is it required? To initialize cuda context? If that's the case we should just call `lazyInitCudaContext` before nvshmem allocs. Also, wouldn't CUDAGuard initialize the context?